pkg/agent/plugin/svidstore/aws/aws.go

```diff
@@ -0,0 +1,240 @@
+	return builtin(New())
+}
+
+func builtin(p *SecretsManagerPlugin) catalog.Plugin {
+	return catalog.MakePlugin(pluginName, svidstore.PluginServer(p))
+}
+
+func New() *SecretsManagerPlugin {
+	return newPlugin(createSecretManagerClient)
+}
+
+func newPlugin(newClient func(secretAccessKey, accessKeyID, region string) (SecretsManagerClient, error)) *SecretsManagerPlugin {
+	p := &SecretsManagerPlugin{}
+	p.hooks.newClient = newClient
+	p.hooks.getenv = os.Getenv
+
+	return p
+}
+
+type secretOptions struct {
+	name     string
+	arn      string
+	kmsKeyID string
+}
+
+func secretFromSelectors(req *svidstore.PutX509SVIDRequest) (*secretOptions, error) {
+	data := svidstore.ParseSelectors(pluginName, req.Selectors)
+
+	opt := &secretOptions{
+		name:     data["secretname"],
+		arn:      data["arn"],
+		kmsKeyID: data["kmskeyid"],
+	}
+
+	if opt.name == "" && opt.arn == "" {
+		return nil, status.Error(codes.InvalidArgument, "secret name or ARN are required")
+	}
+
+	return opt, nil
+}
+
+type Config struct {
+	AccessKeyID     string   `hcl:"access_key_id"`
+	SecretAccessKey string   `hcl:"secret_access_key"`
+	Regions         []string `hcl:"regions"`
+}
+
+type SecretsManagerPlugin struct {
+	svidstore.UnsafeSVIDStoreServer
+
+	log    hclog.Logger
+	config *Config
+	mtx    sync.RWMutex
+
+	hooks struct {
+		newClient func(secretAccessKey, accessKeyID, region string) (SecretsManagerClient, error)
+		getenv    func(string) string
+	}
+}
+
+func (p *SecretsManagerPlugin) SetLogger(log hclog.Logger) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	p.log = log
+}
+
+// Configure configures the SecretsManagerPlugin.
+func (p *SecretsManagerPlugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (*spi.ConfigureResponse, error) {
+	// Parse HCL config payload into config struct
+	config := &Config{}
+	if err := hcl.Decode(config, req.Configuration); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to decode configuration: %v", err)
+	}
+
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	p.config = config
+
+	if config.AccessKeyID == "" {
+		config.AccessKeyID = p.hooks.getenv("AWS_ACCESS_KEY_ID")
+	}
+
+	if config.SecretAccessKey == "" {
+		config.SecretAccessKey = p.hooks.getenv("AWS_SECRET_ACCESS_KEY")
+	}
+
+	return &spi.ConfigureResponse{}, nil
+}
+
+// GetPluginInfo returns the version and other metadata of the plugin.
+func (*SecretsManagerPlugin) GetPluginInfo(context.Context, *spi.GetPluginInfoRequest) (*spi.GetPluginInfoResponse, error) {
+	return &spi.GetPluginInfoResponse{}, nil
+}
+
+// PutX509SVID puts the specified X509-SVID in the configured AWS Secrets Manager
+func (p *SecretsManagerPlugin) PutX509SVID(_ context.Context, req *svidstore.PutX509SVIDRequest) (*svidstore.PutX509SVIDResponse, error) {
+	opt, err := secretFromSelectors(req)
+	if err != nil {
+		return nil, err
+	}
+
+	regions := p.config.Regions
+	// TODO: may we allow default regions to be replaced for selectors?
+	if len(regions) == 0 {
```

Comment:
it is an important TODO, what I want to define here is what we really want to do with regions, on actual code we Put SVIDs in secrets from all `configured` regions from config file, so we are not allowing another regions or put into a single one of those configured regions so, so I'm thinking in different alternatives:

- Keep at it is now.
- Allow to set regions from selectors we may be able to have a selector like: `aws_secretsmanager:region:us-west-1:us-east-1 ` and we can put on those regions. (as default we push to all configured regions or selector regions)
- Get regions from selectors, but it MUST be a subset  of configured regions.